Update app server versions

littleaj opened the pull request:
updates smoke test containers to latest version of their respective app servers.

also, adds readmes to each app server directory for any special instructions.

timja commented on test/smoke/appServers/JBossEAP.6/README.md:
```diff
@@ -0,0 +1,7 @@
+
+Set `JBOSSEAP6_ZIPFILE` to point to the zip file containing JBoss EAP 6.
+
+This is configured to run on Java 7 & 8, so a version >= 6.3.3 is needed; 6.4.0 is reccommended.
```
```suggestion
This is configured to run on Java 7 & 8, so a version >= 6.3.3 is needed; 6.4.0 is recommended.
```

littleaj replied:
good catch, thanks :)

timja commented:
anything holding this up @littleaj or @trask ?

trask commented:
Oh, sorry, I missed this.  Approved!

littleaj commented:
Just running the tests before merge. tomcat changed to sha512sum verification, so I updated that. Also, tomcat 8 is end-of-life so I removed that from the list. tomcat 7 & 8.5 are still being tested.

littleaj commented:
the build is passing; the github integration just doesn't know the difference between failed and cancelled. smoke tests seem to be working after merging in master. I'll merge this when it finishes.
